rundeckapp/grails-spa/packages/ui-trellis/src/library/components/primeVue/PtAutoComplete/PtAutoComplete.vue

```diff
@@ -2,13 +2,22 @@
   <AutoComplete
     v-model="value"
     :suggestions="suggestions"
-    @complete="search"
+    :optionLabel="optionLabel"
+    :name="name"
+    :placeholder="placeholder"
+    :class="{ 'p-invalid': invalid }"
```

Comment:
```suggestion
    :invalid="invalid"
```

The component already has this logic inside of it to add the "p-invalid" class when invalid is true, so all it's required is to pass the prop. If it didn't have this logic, the statement you had added would be completely correct.

Reply:
> The component already has this logic inside of it to add the "p-invalid" class when invalid is true, so all it's required is to pass the prop. If it didn't have this logic, the statement you had added would be completely correct.

For learning and understanding, So If invalid is true then simply passing the prop is enough, and no logic is needed. However, if invalid is false then the logic should be added accordingly?

Reply:
Not exactly. I'm saying that the component inside primevue already has the logic to add the class "p-invalid" when the prop invalid is true, so all it is required is to pass the prop to it. 

If we had a special case and let's say we wanted to add a completely different class, ie "rundeck-autocomplete" when invalid is true, then the logic you had added would be correct (adjusting the class name of course):
`:class="{ 'rundeck-autocomplete': invalid }"`.

Whenever in doubts if the component has the behaviour already, the primevue documentation is super handy: 
https://primevue.org/autocomplete